lambdas/report-user-identity/src/main/java/uk/gov/di/ipv/core/reportuseridentity/ReportUserIdentityHandler.java

```diff
@@ -0,0 +1,196 @@
+                                    usrId ->
+                                            new ReportUserIdentityItem(
+                                                    usrId,
+                                                    Vot.P0.name(),
+                                                    Collections.emptyList(),
+                                                    false))
+                            .toList();
+            reportUserIdentityDataStore.createOrUpdate(reportUserIdentities);
+
+            processUsersToFindLOCAndUpdateDb(userIds);
+
+            LOGGER.info(
+                    LogHelper.buildLogMessage("Completed report processing for user's identity."));
+        } catch (HttpResponseExceptionWithErrorBody | CredentialParseException | ParseException e) {
+            LOGGER.error(
+                    LogHelper.buildErrorMessage(
+                            "Stopped report processing of user's identity.", e));
+        } catch (BatchDeleteException e) {
+            LOGGER.info(LogHelper.buildLogMessage("Error occurred during batch write process."));
+        } finally {
+            LOGGER.info(LogHelper.buildLogMessage("Write output with result summary."));
+            objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
+            objectMapper.writeValue(outputStream, buildReportProcessingResult());
+        }
+    }
+
+    private void processUsersToFindLOCAndUpdateDb(List<String> userIds)
+            throws CredentialParseException, HttpResponseExceptionWithErrorBody, ParseException,
+                    BatchDeleteException {
+        LOGGER.info(
+                LogHelper.buildLogMessage(
+                        String.format("Checking identity for total (%s) users.", userIds.size())));
+        List<ReportUserIdentityItem> reportUserIdentityItems = new ArrayList<>();
+        for (String userId : userIds) {
+            var tacticalVcs = verifiableCredentialService.getVcs(userId);
+            if (!userIdentityService.areVcsCorrelated(tacticalVcs)) {
+                LOGGER.info(
+                        LogHelper.buildLogMessage(
+                                String.format("User (%s) VCs not correlated.", userId)));
+                continue;
+            }
+            boolean migrated = tacticalVcs.stream().allMatch(vc -> vc.getMigrated() != null);
+            var votAttained =
+                    reportUserIdentityService.getStrongestAttainedVotForCredentials(tacticalVcs);
+
+            reportUserIdentityItems.add(
+                    new ReportUserIdentityItem(
+                            userId,
+                            votAttained.orElse(Vot.P0).name(),
+                            reportUserIdentityService.getIdentityConstituent(tacticalVcs),
+                            migrated));
+        }
+        LOGGER.info(LogHelper.buildLogMessage("Updating processed user's identity."));
+        reportUserIdentityDataStore.createOrUpdate(reportUserIdentityItems);
```

Comment:
Note that this `createOrUpdate` has the same semantics as the other `create` methods - it will _overwrite_ existing records, not update them

Reply:
Yeah the item get added if table doesn't contain item with primary key. 
If the table contains an item with the same primary key, it will be replaced with this item.